lib/lhm/entangler.rb

```diff
@@ -98,5 +101,21 @@ def revert
     def strip(sql)
       sql.strip.gsub(/\n */, "\n")
     end
+
+    def with_retry
+      begin
+        retries ||= 0
+        yield
+      rescue StandardError => e
+        if e.message =~ /timeout exceeded/ && retries < MAX_RETRIES
```

Comment:
yeah let me try to make that more specific.  the actual exception we see raised in core is `ActiveRecord::StatementInvalid: Mysql2:Error` so i'll see how that's going to work.